ttnn/cpp/ttnn/operations/pool/pool_utils.cpp

```diff
@@ -1,22 +1,104 @@
+// an argument. It only adds scalars that are different than the scalar preeceding it not to have duplicates of data,
+// this is why we use start and end indices to know how many sequential output elements should be multiplied by the same
+// scalar value.
+std::vector<ScalarInfo> get_bf16_avg_pool_config_scalars(
+    AvgPoolConfig config, uint32_t output_stick_x, uint32_t output_stick_y) {
+    std::vector<ScalarInfo> scalars;
+    float value;
+    bool first_scalar = true;
+    uint32_t last_pool_area = 0;
+
+    if (config.ceil_mode && (config.ceil_w > 0 || config.ceil_h > 0)) {
+        for (uint32_t i = 0; i < config.out_nhw_per_core; i++) {
+            // Compute starting and ending indices of the pooling window
+            int h_start = output_stick_x * config.stride_h - config.pad_h;
+            int w_start = output_stick_y * config.stride_w - config.pad_w;
+            int h_end = std::min(
+                h_start + static_cast<int>(config.kernel_h),
+                static_cast<int>(config.in_h + config.pad_h + config.ceil_h));
+            int w_end = std::min(
+                w_start + static_cast<int>(config.kernel_w),
+                static_cast<int>(config.in_w + config.pad_w + config.ceil_w));
+
+            // Initial pool area
+            int pool_area = (h_end - h_start) * (w_end - w_start);
+
+            // Calculate ceil induced padding overflow beyond input dimensions
+            int pad_h_over = std::max(h_end - static_cast<int>(config.in_h) - static_cast<int>(config.pad_h), 0);
+            int pad_w_over = std::max(w_end - static_cast<int>(config.in_w) - static_cast<int>(config.pad_w), 0);
+
+            // Adjust pool area to exclude padded overflow
+            pool_area -= pad_h_over * config.kernel_w;
+            pool_area -= pad_w_over * config.kernel_h;
+
+            // Re-add intersection if both directions overflowed
+            if (pad_h_over > 0 && pad_w_over > 0) {
+                pool_area += pad_h_over * pad_w_over;
+            }
+
+            // Avoid division by zero
+            pool_area = pool_area > 1 ? pool_area : 1;
+            float value = pool_area > 0 ? 1.f / (float)pool_area : 0.f;
+
+            // Add new scalar if padding config changes
+            if (first_scalar || (uint32_t)pool_area != last_pool_area) {
+                if (!scalars.empty()) {
+                    scalars.back().end = i;
+                }
+                scalars.push_back({i, bfloat16(value).to_packed() << 16, i});
+                first_scalar = false;
+            }
+            last_pool_area = (uint32_t)pool_area;
+
+            // Advance output element coordinates
+            output_stick_y = (output_stick_y + 1) % config.out_w;
+            if (output_stick_y == 0) {
+                output_stick_x = (output_stick_x + 1) % config.out_h;
+            }
+        }
+    } else {
+        value = 1. / (float)(config.kernel_h * config.kernel_w);
```

Comment:
why do we have this codepath, I would rather assert that we are never doing this?